test/unit/kernel/src/KernelStdFunction.cpp

```diff
@@ -7,6 +7,11 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/.
  */
 
+#define TEST_UNIT_KERNEL_KERNEL_STD_FUNCTION
+// clang thinks a macro is unused when only used as the second operand of an &&
+// where the first operand evaluates to false, so we use it here:
+TEST_UNIT_KERNEL_KERNEL_STD_FUNCTION
```

Comment:
Why is this clang workaround not necessary for the other tests?
Should we remove one `KERNEL` from the name?

Reply:
1. Maybe because there the macros are defined via `-D` instead of in the code and are thus maybe not triggering warnings.
2. This one only disables a part of what would be called `TEST_UNIT_KERNEL`, namely the `KERNEL_STD_FUNCTION` part. This is also why this is defined in the source file and not via `target_compile_definitions`.